Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-solution-submit/ContestSolutionSubmitPage.tsx

```diff
@@ -0,0 +1,474 @@
+    const [ getContestById ] = useLazyGetContestByIdQuery();
+    const [ getSubmissionsData ] = useLazyGetSubmissionResultsByProblemQuery();
+    const [ getContestUserParticipation ] = useLazyGetContestUserParticipationQuery();
+
+    const isCompete = participationType === 'compete';
+    const isModalOpen = Boolean(anchorEl);
+
+    const textColorClassName = getColorClassName(themeColors.textColor);
+    const lightBackgroundClassName = getColorClassName(themeColors.baseColor100);
+
+    const {
+        data,
+        isLoading,
+        error,
+    } = useGetContestRegisteredUserQuery({ id: contestId!.toString(), isOfficial: isCompete });
+
+    const { requirePassword, isOnlineExam, name, numberOfProblems, duration, id } = data || {};
+    const {
+        contest,
+        shouldEnterPassword,
+        participantsCount,
+        lastSubmissionTime,
+        userSubmissionsTimeLimit,
+        endDateTimeForParticipantOrContest,
+    } = userContestParticipationData || {};
+    const { problems, allowedSubmissionTypes = [] } = contest || {};
+    const {
+        memoryLimit,
+        timeLimit,
+        fileSizeLimit,
+        checkerName,
+    } = selectedContestDetailsProblem || {};
+
+    const fetchUserParticipationDetails = async () => {
+        setUserParticipationDataLoading(true);
+
+        try {
+            const { data: queryData } = await getContestUserParticipation({
+                id: contestId!,
+                isOfficial: isCompete,
+            });
+
+            setUserParticipationError('');
+            dispatch(setUserContestParticipationData({ participationData: queryData! }));
+            setUserParticipationDataLoading(false);
+        } catch {
+            setUserParticipationError('Error loading user participation data!');
+            dispatch(setUserContestParticipationData({ participationData: null }));
+            setUserParticipationDataLoading(false);
+        }
+    };
+
+    const onStrategyDropdownItemSelect = useCallback((s: any) => {
+        const submissionType = allowedSubmissionTypes.find((type: ISubmissionTypeType) => type.id === s.id);
+
+        setSelectedStrategyValue(s.id);
+        setSelectedSubmissionType(submissionType);
+    }, [ allowedSubmissionTypes ]);
+
+    const strategyDropdownItems = useMemo(
+        () => allowedSubmissionTypes?.map((item: any) => ({ id: item.id, name: item.name })),
+        [ allowedSubmissionTypes ],
+    );
+
+    useEffect(() => {
+        setSubmissionCode('');
+    }, [ selectedContestDetailsProblem ]);
+
+    // fetch contest data, when user has accepted online exam modal,
+    // entered password correctly and has accessed contest data
+    useEffect(() => {
+        if (!userContestParticipationData || userContestParticipationData.contest.id !== Number(contestId)) {
+            fetchUserParticipationDetails();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can you add a comment on why the rule is disabled above this line?

Reply:
done